src/tests/add.rs

```diff
@@ -129,3 +132,103 @@ fn add_key_to_session() {
     assert_eq!(new_payload, new_expected);
     keyring.unlink_key(&key).unwrap();
 }
+
+#[test]
+fn add_encrypted_key_to_session() {
+    let mut keyring = Keyring::attach_or_create(SpecialKeyring::Session).unwrap();
+    let master_key = keyring
+        .add_key::<User, _, _>("foo", "bar".as_bytes())
+        .unwrap();
+    let new_payload = Payload::New {
+        format: Some(Format::default()),
+        keytype: MasterKeyType::User,
+        description: Cow::Borrowed("foo"),
+        keylen: 32,
+    };
+
+    let mut enc_key = keyring
+        .add_key::<Encrypted, _, _>("baz", new_payload)
+        .unwrap();
+
+    // A normal payload update fails
+    assert_eq!(
+        enc_key.update("qux".as_bytes()),
+        Err(errno::Errno(libc::EINVAL))
+    );
+
+    keyring.unlink_key(&enc_key).unwrap();
+    keyring.unlink_key(&master_key).unwrap();
+}
+
+#[test]
+#[should_panic(
+    expected = "called `Result::unwrap()` on an `Err` value: Errno { code: 22, description: Some(\"Invalid argument\") }"
```

Comment:
See #34 for the preferred method for checking for expected errors. The main issue with this strategy is that it doesn't check *which* `unwrap` caused this problem.